### ISSUE TITLE
Activating GPU Mode

### ISSUE THREAD

**Issue by ejunprung** [SkymindIO/SKIL-CE#49]
The problem is that when running the code in Zeppelin, it still runs using CPUs instead of GPUs despite following our GPU mode documentation.

**SKIL Version 1.1.1
DL4J Version 1.0.0-beta
ND4J Version 1.0.0-beta
Supported Spark Version 1.6.3
Supported Scala Version 2.10**

Using Python 2 code with Keras in SKIL Workspaces. Have tried the following: https://docs.skymind.ai/docs/gpu-mode.

**Environment UI**
https://ibb.co/gG3CEe
https://ibb.co/jQnHfK

**Content of the /opt/skil/cuda folder**
deeplearning4j-cuda-9.1-1.0.0-beta.jar 
nd4j-cuda-9.1-1.0.0-beta-linux-x86_64.jar
nd4j-cuda-9.1-1.0.0-beta.jar 
nd4j-native-api-1.0.0-beta.jar

However, CUDA is installed in the default installation folder on the host machine(/usr/local/cuda-9.1/).

**Comment by saudet** [comment]
CUDA doesn't put its libraries in the library path by default, we need to put it in `/etc/ld.so.conf` somewhere as per its instructions. Just creating a link to `/usr/local/cuda` might also work.

With DL4J 1.0.0-beta2 and CUDA 9.2, we can use the "redist" packages instead as well:
https://deeplearning4j.org/docs/latest/deeplearning4j-config-cudnn

**Comment by ShamsUlAzeem** [comment]
Make sure the library paths are set like the following:

```
export PATH=/usr/local/cuda-9.1/bin${PATH:+:${PATH}}
export LD_LIBRARY_PATH=/usr/local/cuda-9.1/lib64${LD_LIBRARY_PATH:+:${LD_LIBRARY_PATH}}
```

Also, you can put the above two lines in /etc/profile.d/cuda91.sh for persistence as: 
`sudo vi /etc/profile.d/cuda91.sh`

and save the file with `:wq`

If that's not the case then I'm assuming the keras is using tensorflow-gpu as the backend. In that case, you might want to install tensorflow-gpu. It's not installed by default. You can do that in the following way:
`/opt/skil/miniconda/bin/conda install tensorflow_gpu`

Then try running your Keras code again.

**Comment by ptamas88** [comment]
Thanks for the answers, this question was asked by me originally.
Strange thing is, i have set the PATH and LD_LIBRARY_PATH environment variables into the ~/.bashrc file but ubuntu hasn't reloaded it since restart :(
I sourced .bashrc and it has been set now.
Other thing is the required conda install, and since I installed it with the command mentioned by @ShamsUlAzeem I can run the Keras code on GPU. Thanks for the advices from both of you! :)
